mlflow/transformers.py

```diff
@@ -245,6 +252,57 @@ def save_model(
                                  the huggingface_hub package must be installed and the version
                                  must be >=0.10.0
 
+    :param inference_config:
+        A dict of valid overrides that can be applied to a pipeline instance during inference.
+        These arguments are used exclusively for the case of loading the model as a ``pyfunc``
+        Model or for use in Spark.
+        These values are not applied to a returned Pipeline from a call to
+        ``mlflow.transformers.load_model()``
+
+        .. Warning:: If the key provided is not compatible with either the
+                  Pipeline instance for the task provided or is not a valid
+                  override to any arguments available in the Model, an
+                  Exception will be raised at runtime. It is very important
+                  to validate the entries in this dictionary to ensure
+                  that they are valid prior to saving or logging.
+
+        An example of providing overrides for a question generation model:
+
+        .. code-block:: python
+
+            from transformers import pipeline, AutoTokenizer
+
+            task = "text-generation"
+            architecture = "gpt2"
+
+            sentence_pipeline = pipeline(
+                task=task, tokenizer=AutoTokenizer.from_pretrained(architecture), model=architecture
+            )
+
+            # Validate that the overrides function
+            prompts = ["Generative models are", "I'd like a coconut so that I can"]
+
+            # validation of config prior to save or log
+            inference_config = {
+                "top_k": 2,
+                "num_beams": 5,
+                "max_length": 30,
+                "temperature": 0.62,
+                "top_p": 0.85,
+                "repetition_penalty": 1.15,
+            }
+
+            # Verify that no exceptions are thrown
+            sentence_generation(prompts, **inference_config)
```

Comment:
Where is `sentence_generation` defined?

Reply:
ah, I had renamed above it to 'sentence_pipeline'. Updated!